File: CODEX/extract_codex_data.js
#!/usr/bin/env node
/**
 * Extract MULVEY_CODEX and FILM_CODEX from codex-uni.html
 * and save as JSON files for codex_universal.html
 */
const fs = require('fs');

// Read the HTML file
const content = fs.readFileSync('codex-uni.html', 'utf8');

// Extract the script content
const scriptMatch = content.match(/<script>([\s\S]*?)<\/script>/);
if (!scriptMatch) {
  console.error('Could not find script tag');
  process.exit(1);
}

const scriptContent = scriptMatch[1];

// Execute the script in a sandbox to get the data
const MULVEY_CODEX = {};
const FILM_CODEX = {};

try {
  // Use eval to execute the const declarations
  eval(scriptContent.match(/const MULVEY_CODEX = \{[\s\S]*?\n  \};/)[0]);
  eval(scriptContent.match(/const FILM_CODEX = \{[\s\S]*?\n  \};/)[0]);
  
  // Save as JSON
  fs.writeFileSync('mulvey_codex.json', JSON.stringify(MULVEY_CODEX, null, 2));
  console.log('✓ Created mulvey_codex.json');
  
  fs.writeFileSync('film_gaze_codex.json', JSON.stringify(FILM_CODEX, null, 2));
  console.log('✓ Created film_gaze_codex.json');
  
  console.log('\nNext steps:');
  console.log('1. These JSON files are now ready for codex_universal.html');
  console.log('2. Select them from the dropdown to load');
  
} catch (error) {
  console.error('Error extracting data:', error.message);
  process.exit(1);
}

File: CODEX/extract_codex_data.py
#!/usr/bin/env python3
"""
Extract MULVEY_CODEX and FILM_CODEX from codex-uni.html
and save as JSON files for codex_universal.html
"""
import re
import json

# Read the HTML file
with open('codex-uni.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract MULVEY_CODEX
mulvey_match = re.search(r'const MULVEY_CODEX = ({.*?});', content, re.DOTALL)
if mulvey_match:
    mulvey_js = mulvey_match.group(1)
    # Convert JS object notation to JSON
    # This is a simple conversion - might need manual tweaking for complex cases
    mulvey_js = re.sub(r"(\w+):", r'"\1":', mulvey_js)  # Add quotes to keys
    mulvey_js = re.sub(r"'", r'"', mulvey_js)  # Replace single quotes with double
    
    try:
        mulvey_data = eval(mulvey_js)  # Using eval since it's our own controlled data
        with open('mulvey_codex.json', 'w', encoding='utf-8') as f:
            json.dump(mulvey_data, f, indent=2, ensure_ascii=False)
        print("✓ Created mulvey_codex.json")
    except Exception as e:
        print(f"Error parsing MULVEY_CODEX: {e}")
        # Fallback: save raw extracted JS
        with open('mulvey_codex_raw.js', 'w') as f:
            f.write(mulvey_js)
        print("  Saved raw JS to mulvey_codex_raw.js for manual conversion")

# Extract FILM_CODEX  
film_match = re.search(r'const FILM_CODEX = ({.*?});', content, re.DOTALL)
if film_match:
    film_js = film_match.group(1)
    # Convert JS object notation to JSON
    film_js = re.sub(r"(\w+):", r'"\1":', film_js)
    film_js = re.sub(r"'", r'"', film_js)
    
    try:
        film_data = eval(film_js)
        with open('film_gaze_codex.json', 'w', encoding='utf-8') as f:
            json.dump(film_data, f, indent=2, ensure_ascii=False)
        print("✓ Created film_gaze_codex.json")
    except Exception as e:
        print(f"Error parsing FILM_CODEX: {e}")
        with open('film_codex_raw.js', 'w') as f:
            f.write(film_js)
        print("  Saved raw JS to film_codex_raw.js for manual conversion")

print("\nNext steps:")
print("1. Verify the JSON files are valid")
print("2. Load them in codex_universal.html using the dropdown")
